internal/service/image/handler.go

```diff
@@ -0,0 +1,115 @@
+package service
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/google/uuid"
+	imageServer "github.com/kubev2v/migration-planner/internal/api/server/image"
+	"github.com/kubev2v/migration-planner/internal/config"
+	"github.com/kubev2v/migration-planner/internal/events"
+	"github.com/kubev2v/migration-planner/internal/image"
+	"github.com/kubev2v/migration-planner/internal/store"
+	"github.com/kubev2v/migration-planner/internal/store/model"
+	"github.com/kubev2v/migration-planner/pkg/metrics"
+	"go.uber.org/zap"
+)
+
+type ImageHandler struct {
+	store       store.Store
+	eventWriter *events.EventProducer
+	cfg         *config.Config
+}
+
+// Make sure we conform to servers Service interface
+var _ imageServer.Service = (*ImageHandler)(nil)
+
+func NewImageHandler(store store.Store, ew *events.EventProducer, cfg *config.Config) *ImageHandler {
+	return &ImageHandler{
+		store:       store,
+		eventWriter: ew,
+		cfg:         cfg,
+	}
+}
+
+func (h *ImageHandler) Health(ctx context.Context, request imageServer.HealthRequestObject) (imageServer.HealthResponseObject, error) {
+	return nil, nil
+}
+
+func (h *ImageHandler) GetImageByToken(ctx context.Context, req imageServer.GetImageByTokenRequestObject) (imageServer.GetImageByTokenResponseObject, error) {
+	writer, ok := ctx.Value(image.ResponseWriterKey).(http.ResponseWriter)
+	if !ok {
+		return imageServer.GetImageByToken500JSONResponse{Message: "error creating the HTTP stream"}, nil
+	}
+
+	if err := image.ValidateToken(ctx, req.Token, h.getSourceKey); err != nil {
+		return imageServer.GetImageByToken401JSONResponse{Message: err.Error()}, nil
+	}
+
+	sourceId, err := image.IdFromJWT(req.Token)
+	if err != nil {
+		return nil, fmt.Errorf("error creating the HTTP stream: %v", err)
+	}
+	source, err := h.getSource(ctx, sourceId)
+	if err != nil {
+		return imageServer.GetImageByToken401JSONResponse{Message: "error creating the HTTP stream"}, nil
+	}
+
+	imageBuilder := image.NewImageBuilder(source.ID)
+	if source.SshPublicKey != nil {
+		imageBuilder = imageBuilder.WithSshKey(*source.SshPublicKey)
+	}
+
+	// TODO: How to add the pull secret???
+	size, err := imageBuilder.Generate(ctx, writer)
```

Comment:
We should add a new option to the builder `WithPullSecret`

Reply:
Yup, but we don't have it...